shashi/jan25.js

```diff
@@ -0,0 +1,47 @@
+
+const isPalindrome = str => {
+  let strNew;
+  if (typeof str === "number") {
+    strNew = str.toString();
+  }
+  if (str.length === 1) {
+    return true;
+  }
+  strNew = (str.split(" ")).join("").toLowerCase();
+
+  let strLen = Math.floor(strNew.length - 1 / 2);
+  for (let i = 0; i < strLen; i++) {
+    if (strNew[i] !== strNew[strNew.length - 1 - i]) {
+      return false
+    }
+  }
+  return true;
+}
+
+//object with Palindrome data
+var palindromeData = {
+  "Repaper": true,
+  "Red rum sir is murder": true,
+  "madam": true,
+  1234567654321: true,
+  12341234: false,
+  "not a plaindrome": false,
+  "a": true,
+  1: true
+};
```

Comment:
not enough test cases. 

"   " this should be false;
"          "  this should be false;
" aaa"
" a a a    "
""
"A a A"
"123 A 321"